### PR TITLE
trying to add time line

### PR DESCRIPTION
**Pull request by mcmenaminadrian**
DO NOT MERGE THIS!

Tell me why it doesn't work instead. PrintATs seem to be messed up, execution stops etc.

### CODE REVIEW

**Comment by mcmenaminadrian** [comment]
OK, worked out that the Spectrum/the emulator is very temperamental when it comes to printing and I had some bad print commands that caused it to break. Fixed that and finally understood you pop/push code and essentially reverted back to what you had.

**Comment by ConorMcPersonal** [comment]
Sorry Adrian, only just seen this

**Comment by mcmenaminadrian** [comment]
It's mergeable now but the protocol should be that the reviewer reviews it and the author merges it - in case the author has spotted anything else etc. Though we can be more relaxed about this - anyway let me know your comments

File: SplitFiles/control.c [hunk 1]
```diff
@@ -51,13 +49,13 @@ void buffer_restack(const ControlBuffer* const buff) {
...
+    for (i = 1; i <= CONTROL_BUFFER_SIZE; i++) {
+        printf(PRINTAT"%c%c""   \0", 28, i);
     }
     // now write out stack
-    for (i = 2; i <= buffer_index; i++) {
+    for (i = 1; i <= buffer_index; i++) {
         command = buffer_getcommand((buff->buffer[i - 1]));
-        printf(PRINTAT"%c%c""%s\0", 21, i, command);
+        printf(PRINTAT"%c%c""%s\0", 28, i, command);
     }
```
Comment (ConorMcPersonal) [inline SplitFiles/control.c]: Quicker to overwrite the stack only once:
for (i = 0; i < buffer_index; ++i) {
    //Write out stack, using fixed-width formatting
    ...
}
for (; i < CONTROL_BUFFER _SIZE; ++i) {
    //Write out spaces, also using fixed-width formatting
    ...
}

the (; above just allows you to use i as it was from the end of the prior loop

Fixed-width formatting looks like %-12s for 12 chars wide

Comment (mcmenaminadrian) [reply SplitFiles/control.c]: This doesn't work! It prints all the empty slots as "EH?" and then drops the top slot

**Pull request merged**